### DISCUSSION TITLE
mpu6050 Can I get the angle? I know this is an algorithmic problem

### DISCUSSION THREAD

**Discussion by bbhxwl** [nanoframework/Home#1135] [Q&A]
I use the Arduino library to get the X angle, not the acceleration and angular velocity.

**Comment by alberk8** [comment]
Short Answer No, because the angle you get from the sensor is over time.   

Long answer.   maybe [this](https://forum.arduino.cc/t/mpu-6050-angle-calculation-from-gyro-sensor-very-inaccurate/672765) and [this](https://www.analog.com/en/app-notes/an-1057.html#:~:text=In%20some%20applications%2C%20where%20the,angle%20of%20tilt%20or%20inclination.)

**Reply by bbhxwl** [reply]
> Short Answer No, because the angle you get from the sensor is over time.
> 
> Long answer. maybe [this](https://forum.arduino.cc/t/mpu-6050-angle-calculation-from-gyro-sensor-very-inaccurate/672765) and [this](https://www.analog.com/en/app-notes/an-1057.html#:~:text=In%20some%20applications%2C%20where%20the,angle%20of%20tilt%20or%20inclination.)

https://github.com/ElectronicCats/mpu6050    I use Arduino to develop. With this library, I can obtain the angle. X is reliable and runs for 2 days. The angle is also accurate. Is there a way to C #?

**Comment by alberk8** [comment]
You can use the code in the sample app to convert to C#.
